test/params_test.c

```diff
@@ -141,9 +141,11 @@ static int raw_set_params(void *vobj, const OSSL_PARAM *params)
             if (!TEST_ptr(obj->p4 = OPENSSL_strndup(params->data,
                                                     params->data_size)))
                 return 0;
+            obj->p4_l = strlen(obj->p4);
         } else if (strcmp(params->key, "p5") == 0) {
             strncpy(obj->p5, params->data, params->data_size);
-            obj->p5_l = strlen(obj->p5) + 1;
+            obj->p5[params->data_size] = '\0';
```

Comment:
Yeah, this needs to be either params->data_size - 1 or ideally min(params->data_size, sizeof(obj->p5) - 1) (and the strncpy above corrected as well.

Reply:
Fixed